pkg/ccl/logictestccl/tests/3node-tenant-multiregion/3node-tenant-multiregion_test: TestTenantExecBuild_distsql_tenant_locality failed

pkg/ccl/logictestccl/tests/3node-tenant-multiregion/3node-tenant-multiregion_test.TestTenantExecBuild_distsql_tenant_locality [failed](https://tanzanite.cluster.engflow.com/invocations/default/0cb8730f-b6f0-4207-85bf-96e5ed3b4f73?testReportRun=8&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQtbXVsdGlyZWdpb246M25vZGUtdGVuYW50LW11bHRpcmVnaW9uX3Rlc3Q=) on release-24.2 @ [509b1e919fa3e89254b4b836854aaddc5c26fc4c](https://github.com/cockroachdb/cockroach/commits/509b1e919fa3e89254b4b836854aaddc5c26fc4c):


```
=== RUN   TestTenantExecBuild_distsql_tenant_locality
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantExecBuild_distsql_tenant_locality1405717197
    test_log_scope.go:81: use -show-logs to present logs inline
[04:59:37] --- done: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant-multiregion/3node-tenant-multiregion_test_/3node-tenant-multiregion_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality with config 3node-tenant-multiregion: 2 tests, 0 failures
    logic.go:4135: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant-multiregion/3node-tenant-multiregion_test_/3node-tenant-multiregion_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality:16: error while processing
    logic.go:4135: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant-multiregion/3node-tenant-multiregion_test_/3node-tenant-multiregion_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality:16: 
        expected success, but found
        (XXUUU) ba: AdminRelocateRange [/Tenant/10/Table/106/1/0/0] RPC error: rpc error: code = Unauthenticated desc = client tenant does not have capability "can_admin_relocate_range" (*kvpb.AdminRelocateRangeRequest)
        transport.go:248: in sendBatch()
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantExecBuild_distsql_tenant_locality1405717197
--- FAIL: TestTenantExecBuild_distsql_tenant_locality (2.07s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantExecBuild_distsql_tenant_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40539

--- a/pkg/sql/logictest/testdata/logic_test/distsql_tenant_locality
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_tenant_locality
@@ -12,8 +12,9 @@ statement ok
 CREATE TABLE t (k INT PRIMARY KEY, v INT, FAMILY (k, v));
 INSERT INTO t SELECT i, i FROM generate_series(1, 6) AS g(i)
 
-# Upreplicate the table's range.
-statement ok
+# Upreplicate the table's range. We need a retry to guarantee that the
+# capability has been picked up.
+statement ok retry
 ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[1, 2, 3], 0)
 
 # Split the ranges in the table.

--- a/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality
+++ b/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality
@@ -12,8 +12,9 @@ user root
 statement ok
 CREATE TABLE t (k INT PRIMARY KEY, v INT, FAMILY (k, v))
 
-# Upreplicate the table's range.
-statement ok
+# Upreplicate the table's range. We need a retry to guarantee that the
+# capability has been picked up.
+statement ok retry
 ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[1, 2, 3], 0)
 
 # Split the ranges in the table.

